lib/Sema/CSDiagnostics.cpp

```diff
@@ -2912,14 +2912,22 @@ bool OutOfOrderArgumentFailure::diagnoseAsError() {
   SourceLoc diagLoc = firstRange.Start;
 
   auto addFixIts = [&](InFlightDiagnostic diag) {
+    // Don't add Fix-Its if one of the ranges is outside of the argument
+    // list, which can happen when we're splicing together an argument list
+    // from multiple sources.
+    auto &SM = getASTContext().SourceMgr;
+    auto argsRange = tuple->getSourceRange();
```

Comment:
When I originally looked at this problem I was thinking about adding a check for `tuple->isImplicit()` and validating ranges there, otherwise make an assert that ranges are correct since for written out call we shouldn't get invalid source ranges for the arguments, WDYT?

Reply:
Yeah, I think pushing the range validation earlier makes sense. We don't have a solid principle around which source locations need to be valid and how they need to relate